### PR TITLE
cluster-init: achieve the Minimum Viable Product

### PR DESCRIPTION
**Pull request by danilo-gemoli**
This PR it's quite extensive and I'm definitely guilty for that. I've been working on it for quite a long time so it would have been hard to split it up in smaller ones.

In a nutshell, these new functionalities have been added:
- integration tests cover the code paths that use an AWS client
- generate `cloudability-agent.yaml` manifests: example [here](https://github.com/openshift/release/blob/a3f6d74e5e77a993e391bc1e69583cff7bb142f6/clusters/build-clusters/build11/cloudability-agent.yaml)
- generate image registry manifests: example [here](https://github.com/openshift/release/blob/8a5567fdd4d0c88637d1f2bf8bbc798fa1cc1e61/clusters/build-clusters/build11/openshift-image-registry)
- generate manifests and secrets for a multiarch cluster
- generate `supplementa-ci-images` manifests: example [here](https://github.com/openshift/release/blob/de1da382a67ae568f77ec6babb8792bf70a66504/clusters/build-clusters/build11/supplemental-ci-images)
- generate `ci-operator` RBAC: example [here](https://github.com/openshift/release/blob/master/clusters/build-clusters/01_cluster/ci-operator/admin_ci-operator_rbac.yaml)
- generate `openshift-ingress-operator` manifests: example [here](https://github.com/openshift/release/blob/f35593a7c88b93aeed0ba2d80679664fa1aa7fb4/clusters/build-clusters/build11/openshift-ingress-operator/ingress-controller-default.yaml)
- improved `MachineSet` manifests generation
- fix a bug on AWS CloudFormation stacks

### CODE REVIEW

File: pkg/clusterinit/onboard/cischedulingwebhook/aws.go [hunk 1]
```diff
@@ -4,27 +4,26 @@ import (
...
+func (ap *awsProvider) GenerateManifests(ctx context.Context, log *logrus.Entry, ci *clusterinstall.ClusterInstall, config *clusterinstall.CISchedulingWebhook) (map[string][]interface{}, error) {
 	manifests := make(map[string][]interface{}, 0)
-	for _, workload := range aw.workloadKeysOrDefault(config.AWS.Workloads) {
-		archToAZ := config.AWS.Workloads[string(workload)]
-		for _, arch := range aw.archsOrDefault(archToAZ) {
-			manifest, err := aw.manifests(ctx, log, ci, string(workload), arch, archToAZ[arch])
+	for _, workload := range util.DefKeys(config.AWS.Workloads, clusterinstall.CIWorkloadDefaults) {
+		archToAZ := config.AWS.Workloads[workload]
+		for _, arch := range util.DefKeys(archToAZ, []string{types.ArchAMD64, types.ArchAARCH64}) {
+			manifest, err := ap.manifests(ctx, log, ci, workload, arch, archToAZ[arch])
```
Comment (deepsm007) [inline pkg/clusterinit/onboard/cischedulingwebhook/aws.go:26]: Are we considering the AZ with eg `1a, 1b, 1c `?

**Comment by danilo-gemoli** [comment]
@deepsm007 https://github.com/openshift/release/pull/58988 should answer you questions

**Pull request merged**